auth/src/main/java/org/aerogear/auth/impl/OIDCTokenAuthenticatorImpl.java

```diff
@@ -1,18 +1,40 @@
 package org.aerogear.auth.impl;
 
+import android.net.Uri;
+
 import org.aerogear.auth.AbstractAuthenticator;
-import org.aerogear.auth.AuthServiceConfig;
 import org.aerogear.auth.AuthenticationException;
+import org.aerogear.auth.IUserPrincipal;
 import org.aerogear.auth.credentials.ICredential;
 import org.aerogear.auth.credentials.OIDCCredentials;
+import org.aerogear.auth.utils.LogoutCompleteHandler;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+import org.aerogear.mobile.core.http.OkHttpServiceModule;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.security.Principal;
 
+import okhttp3.OkHttpClient;
+
+import static org.aerogear.auth.utils.AuthStateManager.*;
+
 /**
  * Authenticates token credentials
  */
 public class OIDCTokenAuthenticatorImpl extends AbstractAuthenticator {
-    public OIDCTokenAuthenticatorImpl(final AuthServiceConfig config) {
+
+    // This depends on how the developers app config will be passed into the SDK.
+    // TODO: Once the above is implemented, this constant will need to be refactored to use the redirect uri chosen by the developer (they may have multiple redirect uri's defined)
+    private static final Uri REDIRECT_URI = Uri.parse("com.feedhenry.securenativeandroidtemplate:/callback");
```

Comment:
I think this url should be a configurable option for the IDM sdk and is provided by the developer

Reply:
At the moment I'm not sure how the developers app config will be passed into the SDK or how it will be managed.  So for now I've left a comment to refactor this once this becomes clearer after the login work is complete.

Do you think its ok to leave it as a constant for now and refactor it once I know how to get this config from the developer?

Reply:
I say we can create a new `AuthConfiguration` class and for now we can set the redirect url there. 

In `AuthService` it should provide a setter for an instance of the `AuthConfiguration`.

`OIDCTokenAuthenticatorImpl` can get the redirect url from the configuration class